[Khoo De Hui] iP

khoodehui opened the pull request.

Commit:
Add Gradle support

Commit:
Level-1

Commit:
Level-2

Commit:
Complete Level-3

Commit:
Complete Level-4

Commit:
Refactor code for Level-4

Commit:
Complete Level-5

Commit:
Complete Level-6

Commit:
Use enum and restructure code

Commit:
Complete  A-TextUiTesting

Commit:
Complete Level-7

Commit:
Complete Level-8

Commit:
Resolve merge conflicts

Commit:
Refactor code for MoreOOP

Commit:
Organise classes into packages

Commit:
Add some JUnit tests

Commit:
Create JAR file

Commit:
Add JavaDocs to main and utility classes

Commit:
Add JavaDocs to task classes

Commit:
Add JavaDocs for command classes

Commit:
Modify code and JavaDocs to comply with coding standards

Commit:
Implement find function

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Resolve merge conflicts

Commit:
Fix bug in find function

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Update gitignore to ignore data files

Commit:
Update .gitignore to ignore data files

Commit:
Remove cached data file

Commit:
Set up Gradle in project

Commit:
Merge branch 'branch-A-Gradle'

Commit:
Add and configure checkstyle

Commit:
Add and configure checkstyle

Commit:
Implement app GUI with JavaFX

Commit:
Implement app GUI

jetnew commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
```
Commenting on coding standard as per iP task, should the JavaDoc for the function be specified here?

MarcTzh commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,85 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + formatTaskTime() + ")";
```
Perhaps you could use String.format instead of manually concatenating the strings.

MarcTzh commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+    }
+    
+    public void markDone() {
+        isDone = true;
```
it would be good to standerdise the usage of this

MarcTzh commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+        StringBuilder output = new StringBuilder();
+        
+        for (int i = 0; i < size(); i++) {
+            output.append(i + 1).append(". ").append(tasks.get(i)).append("\n");
```
perhaps you could split the appends into individual lines for better readibility

MarcTzh reviewed (commented):
Overall well done, LGTM, keep up the good work

hjl99 commented on src/main/java/duke/ui/MainWindow.java:
```diff
@@ -0,0 +1,64 @@
+                DialogBox.getUserDialog(input, userImage),
+                DialogBox.getDukeDialog(response, dukeImage)
+        );
+        if (response.equals("Come back soon!! I'm always bored...")) {
```
Good to consider and add such a scenario in!

hjl99 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+/**
+ * Class the holds the tasks provided by the user.
+ */
+public class TaskList {
```
The class has a clean structure and the methods are following SOLID principles well.

hjl99 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Wrapper class for all types of Command.
+ */
+public abstract class Command {
```
It is nice to see that you have created subclasses of the Command superclass. This guarantees a clean structure as different types of commands can be managed separately. Well done!

hjl99 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+            Command c = Parser.parse(input);
+            c.execute(tasks, ui, storage);
+            return ui.getNextMessage();
+        } catch (DukeException e) {
```
Definitely a nice and clearcut way to manage exceptions!

hjl99 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+        int i = 1;
+        
+        for (Task task : tasks) {
+            if (task.getDescription().contains(keyword)) {
```
Maybe consider if you want to differentiate between lowercases and uppercases? And also, if you wish to search for matching elements in the date and time for events and deadlines as well?

hjl99 reviewed (commented):
I personally find your code pleasant to read as it's structured nicely. Also, your methods seem to follow SOLID principles quite well, and each of them serves its purpose in a very direct and concise manner. Sometimes I might have to scroll up and down for the corresponding classes and functions though, which might be a minor shortcoming. But I guess that's somehow inevitable if a clean coding style is desired.
So overall, well done!

hjl99 commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,47 @@
+     * @throws DukeException If the provided index is not associated with a task.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```
The namings of methods as such are quite clear and direct :)

Commit:
Add assertions

khoodehui committed:
Merge pull request #1 from khoodehui/branch-A-Assertions

Add assertions

Commit:
Refactor code to comply with CheckStyle

Commit:
Refactor Parser and Command classes

Abstract logic for checking command correctness from the Command
classes to the Parser class.

Commit:
Modify JavaDocs

Commit:
Rework storage processes and code improvements

Simplify the save process in the storage class, move serializaton
process to the TaskList, and modify the serialization format. Code
improvements in the various Task classes.

Commit:
Handle additional edge case

Commit:
Improve code

khoodehui committed:
Merge branch 'master' into branch-A-CodeQuality

khoodehui committed:
Merge pull request #2 from khoodehui/branch-A-CodeQuality

Code improvements

Commit:
Refactor code structure

Commit:
Add DateTime and DateTimeParser classes

Commit:
Modify code to use DateTime and DateTimeParser

Commit:
Modify datetime formats

Commit:
Set up CI

Commit:
Create tagging functionality

Commit:
Update JavaDocs

Commit:
Modify load data methods

Commit:
Add remove tag function

Commit:
Add acknowledgement in readme

Commit:
Modify readme format

Commit:
Add user guide and app screenshot

Commit:
Add accepted datetime formats in user guide

Commit:
Fix user guide formatting issue

Commit:
Bug fixes and minor code adjustments

Commit:
Fix used guide formatting issue

Commit:
Customise GUI

Commit:
Modify code to comply with coding standards

khoodehui committed:
Set theme jekyll-theme-cayman
